controllers/reconcilers/kserve_istio_servicemonitor_reconciler.go

```diff
@@ -76,7 +79,12 @@ func (r *KserveIstioServiceMonitorReconciler) Cleanup(ctx context.Context, log l
 	return r.serviceMonitorHandler.DeleteServiceMonitor(ctx, types.NamespacedName{Name: istioServiceMonitorName, Namespace: isvcNs})
 }
 
-func (r *KserveIstioServiceMonitorReconciler) createDesiredResource(isvc *kservev1beta1.InferenceService) (*v1.ServiceMonitor, error) {
+func (r *KserveIstioServiceMonitorReconciler) createDesiredResource(ctx context.Context, isvc *kservev1beta1.InferenceService) (*v1.ServiceMonitor, error) {
```

Comment:
Isn't it enough with the pod monitors?

Also, the match labels is `istio=pilot`. I'd think this will match no pods, unless the ISVC is created on the istio namespace. For the latter case, I *think* (because I haven't checked) it would not fetch metrics from the ISVC, but from istio control plane pods... So, I'm confused about this.

Reply:
it is fetching the metrics correctly.
Doing a quick research: Pods with labels matching the selector criteria are considered Istio-enabled.
It seems it is correct.

Reply:
I understand that it works, yet I still wonder if a `ServiceMonitor` is needed?

This `ServiceMonitor` selects `Services`, not Pods. And this is filtering services having the `istio=pilot` label. On a test cluster, and after deploying an InferenceService, I don't see any `Service` that has such label on the namespace where InferenceService lives.

So, my question still stands: can we stop creating these `ServiceMonitors`?

Reply:
I'll try just with PodMonitor.

In the meantime, @VedantMahabaleshwarkar, what do you think about it?
The ServiceMonitor was there already, I guess it was a suggestion from the istio team, yes?

Reply:
okay, the istiod-system service monitor is not needed, going to remove it.